src/mod_pushoff_fcm.erl

```diff
@@ -0,0 +1,336 @@
+                              {ok, {{_, 200, _}, _, ResponseBody}} ->
+                                    ?DEBUG("+++++ raw response: StatusCode = ~p, Body = ~p", [200, ResponseBody]),
+                                    Timestamp = erlang:timestamp(),
+                                    NewPendingTimer = erlang:send_after(?PENDING_INTERVAL, self(),
+                                                                        {pending_timeout, Timestamp}),
+                                    State#state{out_socket = Socket,
+                                                pending_list = NewPendingList,
+                                                send_queue = NewSendQ,
+                                                pending_timer = NewPendingTimer,
+                                                message_id = NewMessageId,
+                                                pending_timestamp = Timestamp};
+
+                              {ok, {{_, _, _}, _, ResponseBody}} ->
+                                    ?DEBUG("non-recoverable FCM error: ~p, delete registration", [ResponseBody]),
+                                    {NewPendingList1, NewRetryList} = pending_to_retry(NewPendingList, RetryList),
+                                    {NewRetryTimer, Timestamp} = restart_retry_timer(RetryTimer),
+                                    State#state{out_socket = error,
+                                                pending_list = NewPendingList1,
+                                                retry_list = NewRetryList,
+                                                send_queue = NewSendQ,
+                                                retry_timer = NewRetryTimer,
+                                                message_id = NewMessageId,
+                                                retry_timestamp = Timestamp};
+
+                              {error, Reason} ->
+                                    ?ERROR_MSG("FCM request failed: ~p, retrying...", [Reason]),
+                                    {NewPendingList1, NewRetryList} = pending_to_retry(NewPendingList, RetryList),
+                                    {NewRetryTimer, Timestamp} = restart_retry_timer(RetryTimer),
+                                    State#state{out_socket = error,
+                                                pending_list = NewPendingList1,
+                                                retry_list = NewRetryList,
+                                                send_queue = NewSendQ,
+                                                retry_timer = NewRetryTimer,
+                                                message_id = NewMessageId,
+                                                retry_timestamp = Timestamp};
+                              _ -> 
+                                  ?DEBUG("httpc:request() does not matching with any of patterns!!!", [])
```

Comment:
should this be a hard error instead, or just make sure you handle all kinds of patterns?